src/votekit/matrices/boost.py

```diff
@@ -0,0 +1,75 @@
+from ..pref_profile import PreferenceProfile
+from .comentions import comention
+import numpy as np
+from typing import Tuple
+from ._utils import _convert_dict_to_matrix
+from fractions import Fraction
+
+
+def boost_prob(i: str, j: str, pref_profile: PreferenceProfile) -> Tuple[float, float]:
+    """
+    Takes candidates i,j and a preference profile and computes the conditional
+    P(mention i | mention j) and P(mention i). If candidate j is never mentioned,
+    or if the profile has 0 total ballot weight, it returns
+    numpy.nan for the respective probability.
+
+    Args:
+      i (str): Candidate.
+      j (str): Candidate.
+      pref_profile (PreferenceProfile): Profile.
+
+    Returns:
+      tuple[float, float]: P(mention i | mention j), P(mention i)
+    """
+
+    i_mentions = Fraction(0)
+    j_mentions = Fraction(0)
+    both_mentions = Fraction(0)
+
+    for ballot in pref_profile.ballots:
+        if comention(i, ballot):
+            i_mentions += ballot.weight
+
+        if comention(j, ballot):
+            j_mentions += ballot.weight
+
+        if comention([i, j], ballot):
+            both_mentions += ballot.weight
+
+    return (
+        float(both_mentions) / j_mentions if j_mentions else np.nan,
+        (
+            float(i_mentions) / pref_profile.total_ballot_wt
+            if pref_profile.total_ballot_wt
```

Comment:
Again, let's be explicit that we are checking for 0 denominators here

```suggestion
            if (pref_profile.total_ballot_wt != 0)
```